### PR TITLE
Update backup and restore status.

### PR DESCRIPTION
**Pull request by branupama**
This commit defines the backup and restore class status.

### CODE REVIEW

File: include/backup_restore.hpp [hunk 1]
```diff
@@ -12,6 +12,17 @@ namespace vpd
  * @brief class to implement backup and restore VPD.
  *
  */
+
+enum BackupAndRestoreStatus
+{
+    NotStarted = 0,
+    Initiated = 1,
```
Comment (SunnySrivastava1984) [inline include/backup_restore.hpp:19]: construction is basically "Instantiation" so "Initiated -> Instantiated" similarly "InitiationFailed -> InstantiationFailed".
Rest looks good to me.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
Need a proper commit message.
What and why.

File: include/backup_restore.hpp (hunk 1 above)
Comment (branupama) [reply include/backup_restore.hpp]: Updated.

**Comment by SunnySrivastava1984** [comment]
>. This commit defines the backup and restore class status, indicates
the current state of the backup and restore.

I did not understand what is being meant by "backup and restore class status", also what do you mean by "current state of the backup and restore"?

> Change is introduced to avoid calling of backupAndRestore API
multiple times, as it needs to be called only once to backup and
restore VPD keyword’s.

Is this a hard rule? Or the status is only required to get the status of this operation at any given point?

**Review by SunnySrivastava1984 — APPROVED** [review]

File: src/backup_restore.cpp [hunk 2]
```diff
@@ -9,24 +9,43 @@
...
+    BackupAndRestoreStatus::NotStarted;
+
 BackupAndRestore::BackupAndRestore(const nlohmann::json& i_sysCfgJsonObj) :
     m_sysCfgJsonObj(i_sysCfgJsonObj)
 {
-    m_backupAndRestoreCfgJsonObj = jsonUtility::getParsedJson(
-        i_sysCfgJsonObj.value("backupRestoreConfigPath", ""));
+    std::string l_backupAndRestoreCfgFilePath =
+        i_sysCfgJsonObj.value("backupRestoreConfigPath", "");
+    try
```
Comment (jinuthomas) [inline src/backup_restore.cpp:20]: if empty , would a throw be better saying that the file is missing path or path empty in file

File: src/backup_restore.cpp [hunk 3]
```diff
@@ -9,24 +9,43 @@
...
 }
 
 std::tuple<types::VPDMapVariant, types::VPDMapVariant>
     BackupAndRestore::backupAndRestore()
 {
     auto l_emptyVariantPair = std::make_tuple(std::monostate{},
                                               std::monostate{});
-    if (m_backupAndRestoreDone)
+
+    if (m_backupAndRestoreStatus >= BackupAndRestoreStatus::Invoked)
```
Comment (jinuthomas) [inline src/backup_restore.cpp:42]: why is invoke failed considered as invoked ?

File: src/backup_restore.cpp (hunk 2 above)
Comment (jinuthomas) [reply src/backup_restore.cpp:20]: so if you feel you can debug in this format then I am good

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**